garden-backend-service/src/api/dependencies/auth.py

```diff
@@ -0,0 +1,34 @@
+import logging
+from fastapi import Depends, HTTPException
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+
+from src.auth.auth_state import AuthenticationState
+
+
+logger = logging.getLogger("app")
+
+
+def _get_auth_token(
+    authorization: HTTPAuthorizationCredentials = Depends(HTTPBearer(auto_error=False)),
+):
+    """Get the auth token from the Authorization header."""
+    if not authorization:
+        raise HTTPException(status_code=403, detail="Authorization header missing")
+    return authorization.credentials
+
+
+def _get_auth_state(
+    token: str = Depends(_get_auth_token),
+):
+    """Get an AuthenticationState object from the token in the Authorization header."""
+    return AuthenticationState(token)
+
+
+def authenticated(
+    auth_state: AuthenticationState = Depends(_get_auth_state),
```

Comment:
It seems like there's a chain of dependencies here. `authenticated` needs `_get_auth_state`, and `_get_auth_state` needs `_get_auth_token`. But it doesn't seem like the intermediate private ones (`_get_auth_state` and `_get_auth_token`) are used independently anywhere. Is this something framework-y that you need to do or have you just chosen to break it up this way?

Reply:
yeah I actually wanted to get your input on this, we should sync up and take a look at the globus compute code I lifted it from. They seemed to like having the more granular intermediate dependencies (at least for auth stuff) but I wasn't sure if we thought we'd need that granularity

Reply:
they do use all of the intermediate dependencies at least once somewhere iirc